server/monitor/src/main/java/org/apache/accumulo/monitor/Monitor.java

```diff
@@ -612,111 +612,78 @@ public static class CompactionStats {
+  // requests around the same time use the same cached data.
+  private final Supplier<Map<HostAndPort,ScanStats>> tserverScansSupplier =
+      Suppliers.memoizeWithExpiration(this::fetchTServerScans, expirationTimeMinutes, MINUTES);
+
+  private final Supplier<Map<HostAndPort,ScanStats>> sserverScansSupplier =
+      Suppliers.memoizeWithExpiration(this::fetchSServerScans, expirationTimeMinutes, MINUTES);
+
+  private final Supplier<Map<HostAndPort,CompactionStats>> compactionsSupplier =
+      Suppliers.memoizeWithExpiration(this::fetchCompactions, expirationTimeMinutes, MINUTES);
+
+  private final Supplier<ExternalCompactionInfo> compactorInfoSupplier =
+      Suppliers.memoizeWithExpiration(this::fetchCompactorsInfo, expirationTimeMinutes, MINUTES);
+
+  private final Supplier<ExternalCompactionsSnapshot> externalCompactionsSupplier =
+      Suppliers.memoizeWithExpiration(this::computeExternalCompactionsSnapshot,
+          expirationTimeMinutes, MINUTES);
+
   private final RecentLogs recentLogs = new RecentLogs();
-  private final ExternalCompactionInfo ecInfo = new ExternalCompactionInfo();
-
-  private long scansFetchedNanos = System.nanoTime();
-  private long compactsFetchedNanos = System.nanoTime();
-  private long ecInfoFetchedNanos = System.nanoTime();
-  private final long fetchTimeNanos = TimeUnit.MINUTES.toNanos(1);
-  private final long ageOffEntriesMillis = TimeUnit.MINUTES.toMillis(15);
-  // When there are a large amount of external compactions running the list of external compactions
-  // could consume a lot of memory. The purpose of this memoizing supplier is to try to avoid
-  // creating the list of running external compactions in memory per web request. If multiple
-  // request come in around the same time they should use the same list. It is still possible to
-  // have multiple list in memory if one request obtains a copy and then another request comes in
-  // after the timeout and the supplier recomputes the list. The longer the timeout on the supplier
-  // is the less likely we are to have multiple list of external compactions in memory, however
-  // increasing the timeout will make the monitor less responsive.
-  private final Supplier<ExternalCompactionsSnapshot> extCompactionSnapshot =
-      Suppliers.memoizeWithExpiration(() -> computeExternalCompactionsSnapshot(), fetchTimeNanos,
-          TimeUnit.NANOSECONDS);
 
   /**
-   * Fetch the active scans but only if fetchTimeNanos has elapsed.
+   * @return active tablet server scans. Values are cached and refresh after
+   *         {@link #expirationTimeMinutes}.
    */
-  public synchronized Map<HostAndPort,ScanStats> getScans() {
-    if (System.nanoTime() - scansFetchedNanos > fetchTimeNanos) {
-      log.info("User initiated fetch of Active TabletServer Scans");
-      fetchScans();
-    }
-    return Map.copyOf(tserverScans);
+  public Map<HostAndPort,ScanStats> getScans() {
+    return tserverScansSupplier.get();
```

Comment:
It looks like these methods returned an immutable collection previously, but might not be doing that now. Might want to change the fetch* methods to return immutable collections.